docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+The `ImportCommand` will call its own `printResult()` function and return a String message to the `CommandResult` object
+which is then pass back to the `LogicManager`.
+**
+[[search]]
+=== Search feature
+The search function allow users to search for any orders according to the provided input. +
+There are two mode of searching, *general search* or *specific search*. +
+If the user does not provide any `<<command_prefix, ORDER_ATTRIBUTE_PREFIX>>`, a *general search* mode will be performed. +
 
-image::CommitActivityDiagram.png[]
+* *General search* will search for all fields in an order and return orders that have any matching fields. +
 
-==== Design Considerations
+If the user provide any `<<command_prefix, ORDER_ATTRIBUTE_PREFIX>>`, a *specific search* will be performed. +
 
-===== Aspect: How undo & redo executes
+* *Specfic search* will search orders based on the given `<<command_prefix, ORDER_ATTRIBUTE_PREFIX>>`.
 
-* **Alternative 1 (current choice):** Saves the entire address book.
-** Pros: Easy to implement.
-** Cons: May have performance issues in terms of memory usage.
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+==== Class Diagram for Search feature:
 
-===== Aspect: Data structure to support the undo/redo commands
+.Search Command Class Diagram
+image::SearchCommandClassDiagram.svg[]
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+==== Sequence Diagram for Search feature:
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+The sequence diagram for the *Search Command* when searching for any orders that contain *Alice* in any of it's field is shown below:
 
-_{Explain here how the data encryption feature will be implemented}_
+.Search Command Sequence Diagram
+image::SearchCommandSequenceDiagram.svg[]
+
+LogicManager first calls `parseCommand` with arguments representing the user input, `Alice`. The `SearchCommandParser` will then check for any invalid arguments passed by the user. +
```

Comment:
This part just elaborates further on your sequence diagram. I think it would be interesting to discuss your alternative approaches, pros/cons vs your chosen implementation and your justification.